hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/DroolsRulesEngineImpl.java

```diff
@@ -182,6 +182,12 @@ public void fire() {
                     }
                 }
             }
+            long filteringTime = System.currentTimeMillis() - startFiltering;
+            log.debugf("Filtering BatchData [%s] took [%s]", batchData.size(), filteringTime);
+            if (filteringTime > PERF_FILTERING_THRESHOLD) {
+                log.warnf("Filtering Filtering BatchData [%s] took [%s] ms exceeding [%s] ms",
```

Comment:
Duplicate 'Filtering' in message (also see comment about changing 'Filtering' terminology).

Also, not show here in the diff, but above in the code we have this log message:
```
if (log.isDebugEnabled()) {
    log.debug("Deferring more recent " + data + " until older " + previousData + " is " + "processed");
}
```

Perhaps this should be removed completely.  For large amounts of data this may generate too much logging.  If we keep it, please convert to debugf.